// @generated/type-graphql/resolvers/relations/Comments/CommentsRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommentsRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Artworks_1 = require("../../../models/Artworks");
const Comments_1 = require("../../../models/Comments");
const Users_1 = require("../../../models/Users");
const helpers_1 = require("../../../helpers");
let CommentsRelationsResolver = class CommentsRelationsResolver {
    async artworks(comments, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).artworks({});
    }
    async users(comments, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).users({});
    }
};
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Artworks_1.Artworks, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "artworks", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Users_1.Users, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "users", null);
CommentsRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => Comments_1.Comments)
], CommentsRelationsResolver);
exports.CommentsRelationsResolver = CommentsRelationsResolver;
