// @generated/type-graphql/resolvers/relations/Comments/CommentsRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommentsRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Artworks_1 = require("../../../models/Artworks");
const Comments_1 = require("../../../models/Comments");
const CommentsLikes_1 = require("../../../models/CommentsLikes");
const Reports_1 = require("../../../models/Reports");
const Users_1 = require("../../../models/Users");
const CommentsCommentsLikesArgs_1 = require("./args/CommentsCommentsLikesArgs");
const CommentsOther_commentsArgs_1 = require("./args/CommentsOther_commentsArgs");
const CommentsReportsArgs_1 = require("./args/CommentsReportsArgs");
const helpers_1 = require("../../../helpers");
let CommentsRelationsResolver = class CommentsRelationsResolver {
    async users_comments_commentatorIdTousers(comments, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).users_comments_commentatorIdTousers({});
    }
    async users_commentsTousers(comments, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).users_commentsTousers({});
    }
    async artworks(comments, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).artworks({});
    }
    async comments(comments, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).comments({});
    }
    async other_comments(comments, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).other_comments(args);
    }
    async commentsLikes(comments, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).commentsLikes(args);
    }
    async reports(comments, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).comments.findUnique({
            where: {
                id: comments.id,
            },
        }).reports(args);
    }
};
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Users_1.Users, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "users_comments_commentatorIdTousers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Users_1.Users, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "users_commentsTousers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Artworks_1.Artworks, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "artworks", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Comments_1.Comments, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "comments", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Comments_1.Comments], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object, CommentsOther_commentsArgs_1.CommentsOther_commentsArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "other_comments", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [CommentsLikes_1.CommentsLikes], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object, CommentsCommentsLikesArgs_1.CommentsCommentsLikesArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "commentsLikes", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Reports_1.Reports], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Comments_1.Comments, Object, CommentsReportsArgs_1.CommentsReportsArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], CommentsRelationsResolver.prototype, "reports", null);
CommentsRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => Comments_1.Comments)
], CommentsRelationsResolver);
exports.CommentsRelationsResolver = CommentsRelationsResolver;
