// @generated/type-graphql/resolvers/crud/Followers/FollowersCrudResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FollowersCrudResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const graphql_fields_1 = tslib_1.__importDefault(require("graphql-fields"));
const AggregateFollowersArgs_1 = require("./args/AggregateFollowersArgs");
const CreateFollowersArgs_1 = require("./args/CreateFollowersArgs");
const CreateManyFollowersArgs_1 = require("./args/CreateManyFollowersArgs");
const DeleteFollowersArgs_1 = require("./args/DeleteFollowersArgs");
const DeleteManyFollowersArgs_1 = require("./args/DeleteManyFollowersArgs");
const FindFirstFollowersArgs_1 = require("./args/FindFirstFollowersArgs");
const FindManyFollowersArgs_1 = require("./args/FindManyFollowersArgs");
const FindUniqueFollowersArgs_1 = require("./args/FindUniqueFollowersArgs");
const GroupByFollowersArgs_1 = require("./args/GroupByFollowersArgs");
const UpdateFollowersArgs_1 = require("./args/UpdateFollowersArgs");
const UpdateManyFollowersArgs_1 = require("./args/UpdateManyFollowersArgs");
const UpsertFollowersArgs_1 = require("./args/UpsertFollowersArgs");
const helpers_1 = require("../../../helpers");
const Followers_1 = require("../../../models/Followers");
const AffectedRowsOutput_1 = require("../../outputs/AffectedRowsOutput");
const AggregateFollowers_1 = require("../../outputs/AggregateFollowers");
const FollowersGroupBy_1 = require("../../outputs/FollowersGroupBy");
let FollowersCrudResolver = class FollowersCrudResolver {
    async findUniqueFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.findUnique({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async findFirstFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.findFirst({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async findManyFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.findMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async createFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.create({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async createManyFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.createMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async deleteFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.delete({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async updateFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.update({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async deleteManyFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.deleteMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async updateManyFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.updateMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async upsertFollowers(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.upsert({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async aggregateFollowers(ctx, info, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.aggregate({
            ...args,
            ...(0, helpers_1.transformFields)((0, graphql_fields_1.default)(info)),
        });
    }
    async groupByFollowers(ctx, info, args) {
        const { _count, _avg, _sum, _min, _max } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).followers.groupBy({
            ...args,
            ...Object.fromEntries(Object.entries({ _count, _avg, _sum, _min, _max }).filter(([_, v]) => v != null)),
        });
    }
};
tslib_1.__decorate([
    TypeGraphQL.Query(_returns => Followers_1.Followers, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, FindUniqueFollowersArgs_1.FindUniqueFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "findUniqueFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Query(_returns => Followers_1.Followers, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, FindFirstFollowersArgs_1.FindFirstFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "findFirstFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Query(_returns => [Followers_1.Followers], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, FindManyFollowersArgs_1.FindManyFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "findManyFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Mutation(_returns => Followers_1.Followers, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, CreateFollowersArgs_1.CreateFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "createFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, CreateManyFollowersArgs_1.CreateManyFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "createManyFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Mutation(_returns => Followers_1.Followers, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, DeleteFollowersArgs_1.DeleteFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "deleteFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Mutation(_returns => Followers_1.Followers, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, UpdateFollowersArgs_1.UpdateFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "updateFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, DeleteManyFollowersArgs_1.DeleteManyFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "deleteManyFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, UpdateManyFollowersArgs_1.UpdateManyFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "updateManyFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Mutation(_returns => Followers_1.Followers, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, UpsertFollowersArgs_1.UpsertFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "upsertFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Query(_returns => AggregateFollowers_1.AggregateFollowers, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, AggregateFollowersArgs_1.AggregateFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "aggregateFollowers", null);
tslib_1.__decorate([
    TypeGraphQL.Query(_returns => [FollowersGroupBy_1.FollowersGroupBy], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Ctx()),
    tslib_1.__param(1, TypeGraphQL.Info()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Object, Object, GroupByFollowersArgs_1.GroupByFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], FollowersCrudResolver.prototype, "groupByFollowers", null);
FollowersCrudResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => Followers_1.Followers)
], FollowersCrudResolver);
exports.FollowersCrudResolver = FollowersCrudResolver;
