// @generated/type-graphql/enums/ArtworksScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArtworksScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
var ArtworksScalarFieldEnum;
(function (ArtworksScalarFieldEnum) {
    ArtworksScalarFieldEnum["id"] = "id";
    ArtworksScalarFieldEnum["artistId"] = "artistId";
    ArtworksScalarFieldEnum["title"] = "title";
    ArtworksScalarFieldEnum["description"] = "description";
    ArtworksScalarFieldEnum["imageUrl"] = "imageUrl";
    ArtworksScalarFieldEnum["minWorkingHours"] = "minWorkingHours";
    ArtworksScalarFieldEnum["maxWorkingHours"] = "maxWorkingHours";
    ArtworksScalarFieldEnum["minPrice"] = "minPrice";
    ArtworksScalarFieldEnum["maxPrice"] = "maxPrice";
    ArtworksScalarFieldEnum["minHeight"] = "minHeight";
    ArtworksScalarFieldEnum["maxHeight"] = "maxHeight";
    ArtworksScalarFieldEnum["minWidth"] = "minWidth";
    ArtworksScalarFieldEnum["maxWidth"] = "maxWidth";
    ArtworksScalarFieldEnum["createdDate"] = "createdDate";
    ArtworksScalarFieldEnum["deletedDate"] = "deletedDate";
    ArtworksScalarFieldEnum["isDeleted"] = "isDeleted";
})(ArtworksScalarFieldEnum = exports.ArtworksScalarFieldEnum || (exports.ArtworksScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(ArtworksScalarFieldEnum, {
    name: "ArtworksScalarFieldEnum",
    description: undefined,
});
