// @generated/type-graphql/enums/UsersScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
var UsersScalarFieldEnum;
(function (UsersScalarFieldEnum) {
    UsersScalarFieldEnum["id"] = "id";
    UsersScalarFieldEnum["typeId"] = "typeId";
    UsersScalarFieldEnum["firstName"] = "firstName";
    UsersScalarFieldEnum["lastName"] = "lastName";
    UsersScalarFieldEnum["email"] = "email";
    UsersScalarFieldEnum["pass"] = "pass";
    UsersScalarFieldEnum["phone"] = "phone";
    UsersScalarFieldEnum["gender"] = "gender";
    UsersScalarFieldEnum["birthdate"] = "birthdate";
    UsersScalarFieldEnum["address"] = "address";
    UsersScalarFieldEnum["contact"] = "contact";
    UsersScalarFieldEnum["photoUrl"] = "photoUrl";
    UsersScalarFieldEnum["googleUid"] = "googleUid";
    UsersScalarFieldEnum["facebookUid"] = "facebookUid";
    UsersScalarFieldEnum["firebaseToken"] = "firebaseToken";
    UsersScalarFieldEnum["createdDate"] = "createdDate";
    UsersScalarFieldEnum["deletedDate"] = "deletedDate";
    UsersScalarFieldEnum["isDeleted"] = "isDeleted";
})(UsersScalarFieldEnum = exports.UsersScalarFieldEnum || (exports.UsersScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(UsersScalarFieldEnum, {
    name: "UsersScalarFieldEnum",
    description: undefined,
});
