// @generated/type-graphql/resolvers/relations/Artworks/ArtworksRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArtworksRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Artworks_1 = require("../../../models/Artworks");
const Comments_1 = require("../../../models/Comments");
const FavoritesArtworks_1 = require("../../../models/FavoritesArtworks");
const Users_1 = require("../../../models/Users");
const ArtworksCommentsArgs_1 = require("./args/ArtworksCommentsArgs");
const ArtworksFavoritesArtworksArgs_1 = require("./args/ArtworksFavoritesArtworksArgs");
const helpers_1 = require("../../../helpers");
let ArtworksRelationsResolver = class ArtworksRelationsResolver {
    async users(artworks, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).artworks.findUnique({
            where: {
                id: artworks.id,
            },
        }).users({});
    }
    async comments(artworks, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).artworks.findUnique({
            where: {
                id: artworks.id,
            },
        }).comments(args);
    }
    async favoritesArtworks(artworks, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).artworks.findUnique({
            where: {
                id: artworks.id,
            },
        }).favoritesArtworks(args);
    }
};
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Users_1.Users, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Artworks_1.Artworks, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], ArtworksRelationsResolver.prototype, "users", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Comments_1.Comments], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Artworks_1.Artworks, Object, ArtworksCommentsArgs_1.ArtworksCommentsArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], ArtworksRelationsResolver.prototype, "comments", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [FavoritesArtworks_1.FavoritesArtworks], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Artworks_1.Artworks, Object, ArtworksFavoritesArtworksArgs_1.ArtworksFavoritesArtworksArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], ArtworksRelationsResolver.prototype, "favoritesArtworks", null);
ArtworksRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => Artworks_1.Artworks)
], ArtworksRelationsResolver);
exports.ArtworksRelationsResolver = ArtworksRelationsResolver;
