// @generated/type-graphql/resolvers/relations/FavoritesArtworks/FavoritesArtworksRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FavoritesArtworksRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Artworks_1 = require("../../../models/Artworks");
const FavoritesArtworks_1 = require("../../../models/FavoritesArtworks");
const Users_1 = require("../../../models/Users");
const helpers_1 = require("../../../helpers");
let FavoritesArtworksRelationsResolver = class FavoritesArtworksRelationsResolver {
    async users(favoritesArtworks, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).favoritesArtworks.findUnique({
            where: {
                id: favoritesArtworks.id,
            },
        }).users({});
    }
    async artworks(favoritesArtworks, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).favoritesArtworks.findUnique({
            where: {
                id: favoritesArtworks.id,
            },
        }).artworks({});
    }
};
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Users_1.Users, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [FavoritesArtworks_1.FavoritesArtworks, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], FavoritesArtworksRelationsResolver.prototype, "users", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Artworks_1.Artworks, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [FavoritesArtworks_1.FavoritesArtworks, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], FavoritesArtworksRelationsResolver.prototype, "artworks", null);
FavoritesArtworksRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => FavoritesArtworks_1.FavoritesArtworks)
], FavoritesArtworksRelationsResolver);
exports.FavoritesArtworksRelationsResolver = FavoritesArtworksRelationsResolver;
