// @generated/type-graphql/resolvers/relations/Users/UsersRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Artworks_1 = require("../../../models/Artworks");
const ArtworksCollaborators_1 = require("../../../models/ArtworksCollaborators");
const Collections_1 = require("../../../models/Collections");
const Comments_1 = require("../../../models/Comments");
const CommentsLikes_1 = require("../../../models/CommentsLikes");
const FavoritesArtworks_1 = require("../../../models/FavoritesArtworks");
const Followers_1 = require("../../../models/Followers");
const Notifications_1 = require("../../../models/Notifications");
const Reports_1 = require("../../../models/Reports");
const Users_1 = require("../../../models/Users");
const UsersRatings_1 = require("../../../models/UsersRatings");
const UsersArtworksArgs_1 = require("./args/UsersArtworksArgs");
const UsersArtworksCollaboratorsArgs_1 = require("./args/UsersArtworksCollaboratorsArgs");
const UsersCollectionsArgs_1 = require("./args/UsersCollectionsArgs");
const UsersCommentsLikesArgs_1 = require("./args/UsersCommentsLikesArgs");
const UsersComments_commentsTousersArgs_1 = require("./args/UsersComments_commentsTousersArgs");
const UsersComments_comments_commentatorIdTousersArgs_1 = require("./args/UsersComments_comments_commentatorIdTousersArgs");
const UsersFavoritesArtworksArgs_1 = require("./args/UsersFavoritesArtworksArgs");
const UsersFollowersArgs_1 = require("./args/UsersFollowersArgs");
const UsersFollowers_followers_userIdTousersArgs_1 = require("./args/UsersFollowers_followers_userIdTousersArgs");
const UsersNotificationsArgs_1 = require("./args/UsersNotificationsArgs");
const UsersReports_reports_reporterIdTousersArgs_1 = require("./args/UsersReports_reports_reporterIdTousersArgs");
const UsersReports_reports_userIdTousersArgs_1 = require("./args/UsersReports_reports_userIdTousersArgs");
const UsersUsersRatings_usersTousersRatings_qualifierIdArgs_1 = require("./args/UsersUsersRatings_usersTousersRatings_qualifierIdArgs");
const UsersUsersRatings_usersTousersRatings_userIdArgs_1 = require("./args/UsersUsersRatings_usersTousersRatings_userIdArgs");
const helpers_1 = require("../../../helpers");
let UsersRelationsResolver = class UsersRelationsResolver {
    async artworks(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).artworks(args);
    }
    async artworksCollaborators(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).artworksCollaborators(args);
    }
    async collections(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).collections(args);
    }
    async comments_comments_commentatorIdTousers(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).comments_comments_commentatorIdTousers(args);
    }
    async comments_commentsTousers(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).comments_commentsTousers(args);
    }
    async commentsLikes(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).commentsLikes(args);
    }
    async favoritesArtworks(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).favoritesArtworks(args);
    }
    async followers(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).followers(args);
    }
    async followers_followers_userIdTousers(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).followers_followers_userIdTousers(args);
    }
    async notifications(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).notifications(args);
    }
    async reports_reports_reporterIdTousers(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).reports_reports_reporterIdTousers(args);
    }
    async reports_reports_userIdTousers(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).reports_reports_userIdTousers(args);
    }
    async usersRatings_usersTousersRatings_userId(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).usersRatings_usersTousersRatings_userId(args);
    }
    async usersRatings_usersTousersRatings_qualifierId(users, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).users.findUnique({
            where: {
                id: users.id,
            },
        }).usersRatings_usersTousersRatings_qualifierId(args);
    }
};
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Artworks_1.Artworks], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersArtworksArgs_1.UsersArtworksArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "artworks", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [ArtworksCollaborators_1.ArtworksCollaborators], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersArtworksCollaboratorsArgs_1.UsersArtworksCollaboratorsArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "artworksCollaborators", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Collections_1.Collections], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersCollectionsArgs_1.UsersCollectionsArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "collections", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Comments_1.Comments], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersComments_comments_commentatorIdTousersArgs_1.UsersComments_comments_commentatorIdTousersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "comments_comments_commentatorIdTousers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Comments_1.Comments], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersComments_commentsTousersArgs_1.UsersComments_commentsTousersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "comments_commentsTousers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [CommentsLikes_1.CommentsLikes], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersCommentsLikesArgs_1.UsersCommentsLikesArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "commentsLikes", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [FavoritesArtworks_1.FavoritesArtworks], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersFavoritesArtworksArgs_1.UsersFavoritesArtworksArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "favoritesArtworks", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Followers_1.Followers], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersFollowersArgs_1.UsersFollowersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "followers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Followers_1.Followers], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersFollowers_followers_userIdTousersArgs_1.UsersFollowers_followers_userIdTousersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "followers_followers_userIdTousers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Notifications_1.Notifications], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersNotificationsArgs_1.UsersNotificationsArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "notifications", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Reports_1.Reports], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersReports_reports_reporterIdTousersArgs_1.UsersReports_reports_reporterIdTousersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "reports_reports_reporterIdTousers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Reports_1.Reports], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersReports_reports_userIdTousersArgs_1.UsersReports_reports_userIdTousersArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "reports_reports_userIdTousers", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [UsersRatings_1.UsersRatings], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersUsersRatings_usersTousersRatings_userIdArgs_1.UsersUsersRatings_usersTousersRatings_userIdArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "usersRatings_usersTousersRatings_userId", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [UsersRatings_1.UsersRatings], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Users_1.Users, Object, UsersUsersRatings_usersTousersRatings_qualifierIdArgs_1.UsersUsersRatings_usersTousersRatings_qualifierIdArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], UsersRelationsResolver.prototype, "usersRatings_usersTousersRatings_qualifierId", null);
UsersRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => Users_1.Users)
], UsersRelationsResolver);
exports.UsersRelationsResolver = UsersRelationsResolver;
