// @generated/type-graphql/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersRatings = exports.Users = exports.Reports = exports.Notifications = exports.Followers = exports.FavoritesArtworks = exports.CommentsLikes = exports.Comments = exports.Collections = exports.ArtworksTags = exports.ArtworksMovements = exports.ArtworksMaterials = exports.ArtworksColors = exports.ArtworksCollaborators = exports.ArtworksAddresses = exports.Artworks = exports.ArtworkCollections = void 0;
var ArtworkCollections_1 = require("./ArtworkCollections");
Object.defineProperty(exports, "ArtworkCollections", { enumerable: true, get: function () { return ArtworkCollections_1.ArtworkCollections; } });
var Artworks_1 = require("./Artworks");
Object.defineProperty(exports, "Artworks", { enumerable: true, get: function () { return Artworks_1.Artworks; } });
var ArtworksAddresses_1 = require("./ArtworksAddresses");
Object.defineProperty(exports, "ArtworksAddresses", { enumerable: true, get: function () { return ArtworksAddresses_1.ArtworksAddresses; } });
var ArtworksCollaborators_1 = require("./ArtworksCollaborators");
Object.defineProperty(exports, "ArtworksCollaborators", { enumerable: true, get: function () { return ArtworksCollaborators_1.ArtworksCollaborators; } });
var ArtworksColors_1 = require("./ArtworksColors");
Object.defineProperty(exports, "ArtworksColors", { enumerable: true, get: function () { return ArtworksColors_1.ArtworksColors; } });
var ArtworksMaterials_1 = require("./ArtworksMaterials");
Object.defineProperty(exports, "ArtworksMaterials", { enumerable: true, get: function () { return ArtworksMaterials_1.ArtworksMaterials; } });
var ArtworksMovements_1 = require("./ArtworksMovements");
Object.defineProperty(exports, "ArtworksMovements", { enumerable: true, get: function () { return ArtworksMovements_1.ArtworksMovements; } });
var ArtworksTags_1 = require("./ArtworksTags");
Object.defineProperty(exports, "ArtworksTags", { enumerable: true, get: function () { return ArtworksTags_1.ArtworksTags; } });
var Collections_1 = require("./Collections");
Object.defineProperty(exports, "Collections", { enumerable: true, get: function () { return Collections_1.Collections; } });
var Comments_1 = require("./Comments");
Object.defineProperty(exports, "Comments", { enumerable: true, get: function () { return Comments_1.Comments; } });
var CommentsLikes_1 = require("./CommentsLikes");
Object.defineProperty(exports, "CommentsLikes", { enumerable: true, get: function () { return CommentsLikes_1.CommentsLikes; } });
var FavoritesArtworks_1 = require("./FavoritesArtworks");
Object.defineProperty(exports, "FavoritesArtworks", { enumerable: true, get: function () { return FavoritesArtworks_1.FavoritesArtworks; } });
var Followers_1 = require("./Followers");
Object.defineProperty(exports, "Followers", { enumerable: true, get: function () { return Followers_1.Followers; } });
var Notifications_1 = require("./Notifications");
Object.defineProperty(exports, "Notifications", { enumerable: true, get: function () { return Notifications_1.Notifications; } });
var Reports_1 = require("./Reports");
Object.defineProperty(exports, "Reports", { enumerable: true, get: function () { return Reports_1.Reports; } });
var Users_1 = require("./Users");
Object.defineProperty(exports, "Users", { enumerable: true, get: function () { return Users_1.Users; } });
var UsersRatings_1 = require("./UsersRatings");
Object.defineProperty(exports, "UsersRatings", { enumerable: true, get: function () { return UsersRatings_1.UsersRatings; } });
